src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -790,11 +790,15 @@ public boolean isIndexedVersion() {
             return isIndexedVersion = false;
         }
         // If this is the latest published version, we want to confirm that this 
-        // version was successfully indexed after the last publication 
-        
+        // version was successfully indexed after the last publication
+        // We add 3 hours to the indexed time to prevent false negatives
+        // when indexed time gets overwritten in finalizing the publication step
+        // by a value before the release time
+        final long duration = 3 * 60 * 60 * 1000;
+        final Timestamp movedIndexTime = new Timestamp(workingVersion.getDataset().getIndexTime().getTime() + duration);
```

Comment:
Sorry to be a pest, but the null check below may need to happen before this line - ?

Reply:
That was an ugly bug. Thanks for catching it!